spec/outputs/csv_spec.rb

```diff
@@ -7,18 +7,25 @@
 
   subject { described_class.new(options) }
 
-  let(:tmpfile) { Tempfile.new('logstash-spec-output-csv').path }
-  let(:output) { File.readlines(tmpfile) }
-  let(:csv_output) { CSV.read(tmpfile) }
+  let(:tmpfile) { Tempfile.new('logstash-spec-output-csv') }
+  let(:tmpfile_path) { tmpfile.path }
+  let(:output) { File.readlines(tmpfile_path) }
+  let(:csv_output) { CSV.read(tmpfile_path) }
 
   before(:each) do
     subject.register
     subject.multi_receive(events)
   end
 
+  # intentionally close and unlink to make sure the file will have always reference after tmpfile.path
```

Comment:
```suggestion
  # Explicitly close and unlink temp file to ensure immediate cleanup (Tempfile will cleanup eventually)
```
This after is not strictly necessary, but it cant hurt. Either way, i'm not really following the comment. I suggested an alternative the way I understand the intent.